webpack/assets/javascripts/react_app/common/hooks/useSubmitOnEnter.js

```diff
@@ -0,0 +1,24 @@
+import { useEffect } from 'react';
+
+/**
+ * A custom hook that listens for enter on a ref and runs a function on enter keydown
+ * @param  {object} ref a ref created with React's useRef or createRef that scopes the event listener
+ * @param  {function} onSubmit function to execute on enter keydown
+ */
+export const useSubmitOnEnter = (ref, onSubmit) => {
+  const listener = event => {
+    if (event.code === 'Enter' || event.code === 'NumpadEnter') onSubmit();
+  };
+
+  return useEffect(() => {
```

Comment:
I don't see a need to `return` the `useEffect`, since you're not using the return value of `useSubmitOnEnter` anywhere.  For consumers of this hook, what would be the intended use of the return value?  If none is intended, may as well not return anything.  Or think of something useful to return.

Reply:
It's mostly for this eslint rule:
```
/home/vagrant/foreman/webpack/assets/javascripts/react_app/common/hooks/useSubmitOnEnter.js
  15:5  error    Expected an assignment or function call and instead saw an expression                                                    no-unused-expressions
```

Reply:
That looks like a JS error.. which eslint rule?

Reply:
sorry, commented on the wrong thing, I deleted it :)

Reply:
updated to not return the effect :+1: